### PR TITLE
Adding nose2 test framework

### PR DESCRIPTION
**Pull request by stephendiadamo**
Adding a unit test sample file and updating the requirements.txt to add [nose2](https://docs.nose2.io/en/latest/index.html) to the list of requirements.

### CODE REVIEW

File: components/controller_host.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from qunetsim.components import Host
 from utils import DefaultOperationTime
 
+
```
Comment (stephendiadamo) [inline components/controller_host.py:4]: Ah... sorry my IDE auto formatting this file, but I just changed line 91, small syntax issue

File: requirements.txt [hunk 2]
```diff
@@ -1,13 +1,22 @@
 certifi==2020.6.20
+coverage==5.3
```
Comment (stephendiadamo) [inline requirements.txt:2]: Sorry... something is off with this file. I have to change it

**Comment by rheaparekh** [comment]
@stephendiadamo Is this still in progress? Otherwise, I can merge it with my branch and merge everything to the master.

**Comment by stephendiadamo** [comment]
I fixed everything, it's ready to merge.

**Pull request merged**